canonical-web-and-design/maas-ui
# Fix duplicates in ngRepeat.

## Done

- Remove duplicate messages in `confirmationDetails` and duplicate IPs in subnets. 

A function for removing duplicates was already present in `machines_table` (clearly someone has attempted to solve this problem for a different ngRepeat), so I've moved that to an exported filter.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- visit the networking details tab for a controller, ensure everything still works!

## Fixes

Fixes: #1439 

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
